### ISSUE TITLE
Error if creating a typed struct in JRuby

### ISSUE THREAD

**Issue by Khalasar** [sorbet/sorbet#1188]
#### Input

```ruby
require "sorbet-runtime"

class Foo < T::Struct
  const :test, String
end

Foo.new(test: "test")
```

#### Observed output

```
NoMethodError: super: no superclass method `valid_props' for #<T::Props::Decorator:0x1e4a7dd4 @class=Foo>
```

#### Used Versions
**JRuby:** 9.2.7.0
**sorbet-runtime:** 0.4.4379
- - -

In Ruby 2.6.3 it is working for me.

**Comment by wealth** [comment]
A little different error, but still could not use T::Struct.

**sorbet-runtime:** 0.5.5226
**JRuby:** 9.2.9.0

```
➜ irb
jruby-9.2.9.0 :001 > require "sorbet-runtime"
 => true 
jruby-9.2.9.0 :002 > 
jruby-9.2.9.0 :003 > class Foo < T::Struct
jruby-9.2.9.0 :004?>     const :test, String
jruby-9.2.9.0 :005?>   end
Traceback (most recent call last):
       16: from /home/prog/.rvm/gems/jruby-9.2.9.0/gems/sorbet-runtime-0.5.5226/lib/types/props/_props.rb:140:in `const'
       15: from /home/prog/.rvm/gems/jruby-9.2.9.0/gems/sorbet-runtime-0.5.5226/lib/types/props/_props.rb:109:in `prop'
       14: from /home/prog/.rvm/gems/jruby-9.2.9.0/gems/sorbet-runtime-0.5.5226/lib/types/private/methods/_methods.rb:240:in `block in prop_defined'
       13: from org/jruby/RubyMethod.java:132:in `call'
       12: from /home/prog/.rvm/gems/jruby-9.2.9.0/gems/sorbet-runtime-0.5.5226/lib/types/props/decorator.rb:430:in `prop_defined'
       11: from /home/prog/.rvm/gems/jruby-9.2.9.0/gems/sorbet-runtime-0.5.5226/lib/types/props/serializable.rb:317:in `prop_validate_definition!'
       10: from /home/prog/.rvm/gems/jruby-9.2.9.0/gems/sorbet-runtime-0.5.5226/lib/types/props/optional.rb:55:in `prop_validate_definition!'
        9: from /home/prog/.rvm/gems/jruby-9.2.9.0/gems/sorbet-runtime-0.5.5226/lib/types/private/methods/_methods.rb:240:in `block in prop_validate_definition!'
        8: from org/jruby/RubyMethod.java:132:in `call'
        7: from /home/prog/.rvm/gems/jruby-9.2.9.0/gems/sorbet-runtime-0.5.5226/lib/types/props/decorator.rb:295:in `prop_validate_definition!'
        6: from org/jruby/RubyArray.java:4511:in `any?'
        5: from /home/prog/.rvm/gems/jruby-9.2.9.0/gems/sorbet-runtime-0.5.5226/lib/types/props/decorator.rb:295:in `block in prop_validate_definition!'
        4: from /home/prog/.rvm/gems/jruby-9.2.9.0/gems/sorbet-runtime-0.5.5226/lib/types/props/serializable.rb:282:in `valid_rule_key?'
        3: from /home/prog/.rvm/gems/jruby-9.2.9.0/gems/sorbet-runtime-0.5.5226/lib/types/private/methods/_methods.rb:240:in `block in valid_rule_key?'
        2: from org/jruby/RubyMethod.java:132:in `call'
        1: from /home/prog/.rvm/gems/jruby-9.2.9.0/gems/sorbet-runtime-0.5.5226/lib/types/props/pretty_printable.rb:23:in `valid_rule_key?'
NoMethodError (super: no superclass method `valid_rule_key?' for #<T::Props::Decorator:0x76508ed1 @class=Foo>)
jruby-9.2.9.0 :006 >
```

**Comment by st0012** [comment]
I can confirm @wealth's case, but I suspect this is a jruby bug and maybe need a special workaround?


### JRuby 9.2.13

<img width="826" alt="ruby2 6 5" src="https://user-images.githubusercontent.com/5079556/94280836-79a56280-ff80-11ea-8862-9891fa8a2fcb.png">

### Ruby 2.6.5

<img width="820" alt="jruby-error" src="https://user-images.githubusercontent.com/5079556/94280872-845ff780-ff80-11ea-8caa-3457b2482ea1.png">

not trying to push the maintainers here, but I really like sorbet and I'm trying to adopt it in the [Sentry sdk](https://github.com/getsentry/sentry-ruby)'s following major release (https://github.com/getsentry/sentry-ruby/pull/1040). and if it can't work on jruby I'll have to drop this idea 😢

**Comment by jez** [comment]
sorbet-runtime is normal Ruby code. If it works in Ruby but doesn't work in JRuby, that's a bug in JRuby.

**Comment by headius** [comment]
> If it works in Ruby but doesn't work in JRuby, that's a bug in JRuby.

In the future it would be helpful if someone would file JRuby issues for things that are obviously bugs in JRuby. @ashkulz finally did so five days ago (over two years after this report), and that is the first I ever heard of this bug. Had it been filed in 2020, we could have fixed it in 2020.

**Comment by headius** [comment]
This appears to be due to a behavior change in `bind` and `bind_call` that makes it possible to `super` from a rebound module method. This behavior did not exist prior to Ruby 2.7 and was added at the same time as `bind_call`. The behavior change was never documented and no tests were introduced for it.

See https://github.com/jruby/jruby/pull/7551.

**Comment by ashkulz** [comment]
Good news -- that bug is now fixed and should show up in the next JRuby 9.3 / 9.4 release :tada:
